holoviews/tests/operation/test_downsample.py

```diff
@@ -18,3 +24,27 @@ def test_downsample1d_multi(plottype):
     for n in figure_values:
         for value in n.data.values():
             assert value.size == downsample1d.width
+
+
+@pytest.mark.parametrize("algorithm", _ALGORITHMS.values(), ids=_ALGORITHMS)
```

Comment:
I think the way these two tests are written leads to repeating the same tests (non-`tsdownsample` tests run twice) and skipping tests when `tsdownsample` is unimported while it should be a success (in `test_downsample_algorithm`)?

This is maybe more explicit?

```py
TSD_ALGOS = {k: v for k, v in _ALGORITHMS.items() if k in ['minmax', 'minmax-lttb', 'm4']}

@pytest.mark.parametrize("algorithm", TSD_ALGOS.values(), ids=TSD_ALGOS)
def test_downsample_algorithm(algorithm, unimport):
    unimport("tsdownsample")
    x = np.arange(1000)
    y = np.random.rand(1000)
    width = 20
    with pytest.raises(NotImplementedError):
        algorithm(x, y, width)
```

Reply:
There is an overlap with lttb which works with and without tsdownsample. 

I originally had it passing, but liked I got feedback that the test was skipped for tsdownsample only functions. 

I think there is a lot of ways to do this. I preferred running some test twice than using logic as it could get out of sync. If you much more prefer it your way I will update the tests.

Reply:
Ok I see. The only problem I have then with the first test is that I can comment out `unimport("tsdownsample")` and will get no error with and without `tsdownsample` installed, it'll either skip or succeed.

Reply:
I agree with you that it is not ideal. I think it is okay, as your scenario changes the behavior of the unit test itself. 

I will gladly listen if you have a proposal on how to test it.

Reply:
Ok yes that's fine by me.